src/core/toga/widgets/switch.py

```diff
@@ -9,85 +11,156 @@ class Switch(Widget):
         and False (off, unchecked).
 
     Args:
-        label (str): Text to be shown next to the switch.
+        text (str): Text to be shown next to the switch.
         id (str): AN identifier for this widget.
         style (:obj:`Style`): An optional style object.
             If no style is provided then a new one will be created for the widget.
-        on_toggle (``callable``): Function to execute when pressed.
-        is_on (bool): Current on or off state of the switch.
+        on_change (``callable``): Function to execute when pressed.
+        value (bool): Current on or off state of the switch.
         enabled (bool): Whether or not interaction with the button is possible, defaults to `True`.
         factory (:obj:`module`): A python module that is capable to return a
             implementation of this class with the same name. (optional & normally not needed)
     """
 
     def __init__(
             self,
-            label,
+            text,
             id=None,
             style=None,
-            on_toggle=None,
-            is_on=False,
+            on_change=None,
+            on_toggle=None,  # DEPRECATED!
+            is_on=None,
+            value=False,
             enabled=True,
             factory=None,
     ):
         super().__init__(id=id, style=style, factory=factory)
 
         self._impl = self.factory.Switch(interface=self)
 
-        self.label = label
-        self.on_toggle = on_toggle
-        self.is_on = is_on
+        self.text = text
```

Comment:
It would be fairly straightforward to add support for a named `label=` argument in the constructor,  so I'd argue we should add this as well. Yes, it means introducing a "temporary" keyword argument, but it's also an easy thing to delete in future. We can also re-group the arguments so there's a single block of "deprecated" arguments at the end.